compiler/test/fail_compilation/shared.d

```diff
@@ -87,7 +87,6 @@ fail_compilation/shared.d(2154): Error: direct access to shared `*c.c1` is not a
 fail_compilation/shared.d(2160): Error: direct access to shared `*c.c1.c1` is not allowed, see `core.atomic`
 fail_compilation/shared.d(2181): Error: direct access to shared `k` is not allowed, see `core.atomic`
 fail_compilation/shared.d(2187): Error: direct access to shared `k.k2.k1` is not allowed, see `core.atomic`
-fail_compilation/shared.d(2194): Error: direct access to shared `(new shared(K2)).k1` is not allowed, see `core.atomic`
```

Comment:
Because your code example returns a compile-time known pointer to `c`, while the code in the test case would return the shared pointer value of `K2` + the offset of `k1`.

Reply:
I think it could be allowed on the grounds that `new shared (K2)` is isolated (no aliases).

Reply:
Ok, are there any changes that you require for this PR?

Reply:
Someone wrote code and a test explicitly for `(new shared(K2)).k1`, this bug fix should not introduce new semantics to `shared` without approval.

Reply:
Also, as noted in bugzilla there's two issues:
- cannot create shared class instance
- cannot use `synchronized`
This PR should either fix both, or rename the issue to the one that it fixes.